# Support Vector Machines (SVM) Warmup

Using the iris dataset built into sklearn (see here: https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_iris.html), train a vanilla (all default parameters) SVM model with 'train_test_split' and print out it's accuracy score. Make sure to use random_state=107 for the split and model. Use a test size of 30% and stratify the split on y. Use Species as the target variable and use all other variables as features except for Id Share your accuracy score in the thread and explain its interpretation.

In [3]:
from sklearn.datasets import load_iris

In [31]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt  
%matplotlib inline
import seaborn as sns 
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import classification_report, confusion_matrix , accuracy_score
from sklearn.model_selection import train_test_split  
from sklearn.model_selection import cross_validate
from sklearn.svm import SVC

In [22]:
dataset=load_iris()
data=pd.DataFrame(dataset['data'],columns=['Petal length','Petal Width','Sepal Length','Sepal Width'])
data['Species']=dataset['target']
data['Species']=data['Species'].apply(lambda x: dataset['target_names'][x])
data.head()

Petal length  Petal Width  Sepal Length  Sepal Width Species
0           5.1          3.5           1.4          0.2  setosa
1           4.9          3.0           1.4          0.2  setosa
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa

In [24]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
Petal length    150 non-null float64
Petal Width     150 non-null float64
Sepal Length    150 non-null float64
Sepal Width     150 non-null float64
Species         150 non-null object
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


In [25]:
X = data.drop('Species', axis=1)  
y = data['Species'] 
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.30, random_state = 107, stratify=y) 

In [29]:
ss = StandardScaler()
X_train = ss.fit_transform(X_train)
X_test = ss.transform(X_test)

In [33]:
model=SVC()

In [35]:
model.fit(X_train, y_train)

SVC(C=1.0, cache_size=200, class_weight=None, coef0=0.0,
    decision_function_shape='ovr', degree=3, gamma='auto_deprecated',
    kernel='rbf', max_iter=-1, probability=False, random_state=None,
    shrinking=True, tol=0.001, verbose=False)

In [37]:
pred=model.predict(X_test)

In [39]:
print(confusion_matrix(y_test,pred))

[[14  1  0]
 [ 0 14  1]
 [ 0  1 14]]


In [38]:
print(classification_report(y_test, pred))

              precision    recall  f1-score   support

      setosa       1.00      0.93      0.97        15
  versicolor       0.88      0.93      0.90        15
   virginica       0.93      0.93      0.93        15

    accuracy                           0.93        45
   macro avg       0.94      0.93      0.93        45
weighted avg       0.94      0.93      0.93        45



In [41]:
print(accuracy_score(y_test,pred))

0.9333333333333333


**Accuracy = (correctly predicted class / total testing class) × 100<br>
So an Accuracy score of 93.33 means that the model predicted 93.33% of the test data classes correctly.**